zcash_primitives/src/transaction/builder.rs

```diff
@@ -1745,4 +1946,172 @@ mod tests {
+                    AssetBase::native(),
+                    Memo::Empty.encode().into_bytes(),
+                )
+                .unwrap();
+            let (bundle, meta) = orchard_builder
+                .build::<i64, OrchardVanilla>(&mut rng)
+                .unwrap();
+            let action = bundle
+                .actions()
+                .get(meta.output_action_index(0).unwrap())
+                .unwrap();
+            let domain = OrchardDomain::for_action(action);
+            let (note, _, _) = try_note_decryption(&domain, &ivk.prepare(), action).unwrap();
+            note
+        };
+        let (anchor, merkle_path) = {
+            let cmx: orchard::note::ExtractedNoteCommitment = note.commitment().into();
+            let leaf = MerkleHashOrchard::from_cmx(&cmx);
+            let mut tree = ShardTree::<_, 32, 16>::new(
+                MemoryShardStore::<MerkleHashOrchard, u32>::empty(),
+                100,
+            );
+            tree.append(leaf, incrementalmerkletree::Retention::Marked)
+                .unwrap();
+            tree.checkpoint(9_999_999).unwrap();
+            let position = 0.into();
+            let merkle_path = tree
+                .witness_at_checkpoint_depth(position, 0)
+                .unwrap()
+                .unwrap();
+            let anchor = merkle_path.root(leaf);
+            (anchor.into(), merkle_path.into())
+        };
+
+        let build_config = BuildConfig::TxV6 {
+            sapling_anchor: Some(sapling::Anchor::empty_tree()),
+            orchard_anchor: Some(anchor),
+        };
+        let mut builder = Builder::new(TEST_NETWORK, tx_height, build_config);
+
+        // Pick two different asset descriptions and derive their asset bases
+        let asset_desc_hash_1 =
+            compute_asset_desc_hash(&NonEmpty::from_slice(b"This is Asset 1").unwrap());
+        let asset_desc_hash_2 =
+            compute_asset_desc_hash(&NonEmpty::from_slice(b"This is Asset 2").unwrap());
+
+        // Create an asset creation function, to simulate the output from querying global state,
+        // under the assumption that only the assets in prev_issued_assets have been issued before,
+        // and no other assets are previously issued.
+        fn is_asset_newly_created(asset: AssetBase, prev_issued_assets: &[AssetBase]) -> bool {
+            if asset == AssetBase::native() {
+                return false;
+            }
+            !prev_issued_assets.contains(&asset)
+        }
+
+        let issue_info = IssueInfo {
+            recipient,
+            value: NoteValue::from_raw(10),
+        };
+
+        // Add Orchard spend and output for the fees (spending existing Orchard note).
+        builder
+            .add_orchard_spend::<zip317::FeeRule>(fvk, note, merkle_path)
+            .unwrap();
+        builder
+            .add_orchard_output::<zip317::FeeRule>(
+                Some(ovk),
+                recipient,
+                9_475_000,
```

Comment:
Use named constants instead of magic numbers. Also consider a value that does not raise questions (maybe simply 1).

Reply:
Done